### PR TITLE
Update schneider_electric.ts

### PR DESCRIPTION
**Pull request by TGruber-HH**
added MEG5126-0300_MEG5152-0000

### CODE REVIEW

File: src/devices/schneider_electric.ts [hunk 1]
```diff
@@ -1293,6 +1293,16 @@ const definitions: Definition[] = [
             await endpoint1.read('hvacUserInterfaceCfg', ['keypadLockout', 'tempDisplayMode']);
         },
     },
+        {
+        zigbeeModel: ['2GANG/ESWITCH/2'],
+        model: 'MEG5126-0300/MEG5152-0000',
+        vendor: 'Schneider Electric',
+        description: 'Merten MEG5152 switch insert (2fold) with Merten System M push button (2fold)',
+        extend: [deviceEndpoints({"endpoints":{"1":1,"2":2,"21":21,"22":22}}), identify(), onOff({"powerOnBehavior":false,"endpointNames":["1","2"]}), commandsOnOff({"endpointNames":["21","22"]}), commandsLevelCtrl({"endpointNames":["21","22"]}), commandsWindowCovering({"endpointNames":["21","22"]})],
```
Comment (Koenkk) [inline src/devices/schneider_electric.ts:1301]: Can you use more friendly endpoint names here? E.g. `"left":1`

Comment (Koenkk) [inline src/devices/schneider_electric.ts:1301]: Can you use more friendly endpoint names here? E.g. `"left":1`

Comment (Koenkk) [inline src/devices/schneider_electric.ts:1301]: Could you add the command here? example: https://github.com/Koenkk/zigbee-herdsman-converters/blob/c400db4fb3ecd68e56c9ed06c4cf66126c007c4c/src/devices/ikea.ts#L802

File: src/devices/schneider_electric.ts [hunk 2]
```diff
@@ -1293,6 +1293,20 @@ const definitions: Definition[] = [
             await endpoint1.read('hvacUserInterfaceCfg', ['keypadLockout', 'tempDisplayMode']);
         },
     },
+    {
+        zigbeeModel: ['2GANG/ESWITCH/2'],
+        model: 'MEG5126-0300/MEG5152-0000',
+        vendor: 'Schneider Electric',
+        fromZigbee: [fz.on_off, fz.command_on, fz.command_off],
```
Comment (Koenkk) [inline src/devices/schneider_electric.ts:1300]: Does it also work without `toZigbee` and `fromZigbee`?

File: src/devices/schneider_electric.ts [hunk 3]
```diff
@@ -1293,6 +1293,20 @@ const definitions: Definition[] = [
...
+        toZigbee: [tz.on_off],
+        description: 'Merten MEG5152 switch insert (2fold) with Merten System M push button (2fold)',
+        extend: [deviceEndpoints({'endpoints': {'left': 1, 'right': 2, 'left_sw': 21, 'right_sw': 22}}), identify(),
+            onOff({'powerOnBehavior': false, 'endpointNames': ['left', 'right']}),
+            commandsOnOff({'endpointNames': ['left_sw', 'right_sw']}),
+        ],
+        meta: {multiEndpoint: true},
```
Comment (Koenkk) [inline src/devices/schneider_electric.ts:1307]: This line can also be removed.

**Comment by TGruber-HH** [comment]
YES it is working without fz. and tz.

 

BR

Thomas

 
 

Gesendet: Samstag, 13. April 2024 um 09:39 Uhr
Von: "Koen Kanters" ***@***.***>
An: "Koenkk/zigbee-herdsman-converters" ***@***.***>
Cc: "Tommy1968" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [Koenkk/zigbee-herdsman-converters] Update schneider_electric.ts (PR #7385)


 

@Koenkk commented on this pull request.


In src/devices/schneider_electric.ts:

> @@ -1293,6 +1293,20 @@ const definitions: Definition[] = [
             await endpoint1.read('hvacUserInterfaceCfg', ['keypadLockout', 'tempDisplayMode']);
         },
     },
+    {
+        zigbeeModel: ['2GANG/ESWITCH/2'],
+        model: 'MEG5126-0300/MEG5152-0000',
+        vendor: 'Schneider Electric',
+        fromZigbee: [fz.on_off, fz.command_on, fz.command_off],


Does it also work without toZigbee and fromZigbee?

**Comment by TGruber-HH** [comment]
1) shall I remove it in the local github and initiate the Pull again?

 

2) shall I remove the fz. and tz. in local github and initiate Pull request, too ?

 

BR

Thomas

 
 

Gesendet: Samstag, 13. April 2024 um 09:39 Uhr
Von: "Koen Kanters" ***@***.***>
An: "Koenkk/zigbee-herdsman-converters" ***@***.***>
Cc: "Tommy1968" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [Koenkk/zigbee-herdsman-converters] Update schneider_electric.ts (PR #7385)


 

@Koenkk commented on this pull request.


In src/devices/schneider_electric.ts:

> @@ -1293,6 +1293,20 @@ const definitions: Definition[] = [
             await endpoint1.read('hvacUserInterfaceCfg', ['keypadLockout', 'tempDisplayMode']);
         },
     },
+    {
+        zigbeeModel: ['2GANG/ESWITCH/2'],
+        model: 'MEG5126-0300/MEG5152-0000',
+        vendor: 'Schneider Electric',
+        fromZigbee: [fz.on_off, fz.command_on, fz.command_off],
+        toZigbee: [tz.on_off],
+        description: 'Merten MEG5152 switch insert (2fold) with Merten System M push button (2fold)',
+        extend: [deviceEndpoints({'endpoints': {'left': 1, 'right': 2, 'left_sw': 21, 'right_sw': 22}}), identify(),
+            onOff({'powerOnBehavior': false, 'endpointNames': ['left', 'right']}),
+            commandsOnOff({'endpointNames': ['left_sw', 'right_sw']}),
+        ],
+        meta: {multiEndpoint: true},


This line can also be removed.

**Comment by Koenkk** [comment]
Great, could you update this PR?

**Comment by TGruber-HH** [comment]
Dear Koen,

 

#7385 has been updated few seconds ago

 

 

 

GREAT JOB for the Zigbee2Mqtt, I really like the project - respect of your performance

 

Best regards

Thomas

 
 

Gesendet: Montag, 15. April 2024 um 21:45 Uhr
Von: "Koen Kanters" ***@***.***>
An: "Koenkk/zigbee-herdsman-converters" ***@***.***>
Cc: "Tommy1968" ***@***.***>, "Author" ***@***.***>
Betreff: Re: [Koenkk/zigbee-herdsman-converters] Update schneider_electric.ts (PR #7385)


 

Great, could you update this PR?

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**